<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1031</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QFrame" name="row_1">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_19">
       <item>
        <widget class="QFrame" name="frame_div_content_1">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>110</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>110</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_17">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QFrame" name="frame_content_wid_1">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <layout class="QGridLayout" name="gridLayout">
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="0" column="1">
                <widget class="QLineEdit" name="lineedit_imported_filename">
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(33, 37, 43);</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="placeholderText">
                  <string>Type here</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QPushButton" name="btn_open_local_file">
                 <property name="minimumSize">
                  <size>
                   <width>150</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="font">
                  <font>
                   <family>Segoe UI</family>
                   <pointsize>10</pointsize>
                   <italic>false</italic>
                   <bold>false</bold>
                  </font>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(52, 59, 72);</string>
                 </property>
                 <property name="text">
                  <string>Open</string>
                 </property>
                 <property name="icon">
                  <iconset resource="resources.qrc">
                   <normaloff>:/icons/images/icons/cil-folder-open.png</normaloff>:/icons/images/icons/cil-folder-open.png</iconset>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QPushButton" name="btn_FeatureProcessing">
                 <property name="minimumSize">
                  <size>
                   <width>150</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="font">
                  <font>
                   <family>Segoe UI</family>
                   <pointsize>10</pointsize>
                   <italic>false</italic>
                   <bold>false</bold>
                  </font>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(52, 59, 72);</string>
                 </property>
                 <property name="text">
                  <string>FeatureProcessing</string>
                 </property>
                 <property name="icon">
                  <iconset resource="resources.qrc">
                   <normaloff>:/icons/images/icons/cil-data-process.png</normaloff>:/icons/images/icons/cil-data-process.png</iconset>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_title_wid_1">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>50</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QLabel" name="label_missing_values_handling">
                 <property name="text">
                  <string>缺失值处理：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="combobox_missing_values_handling">
                 <item>
                  <property name="text">
                   <string>平均值填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>中位数填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>众数填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>固定值填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>最近邻填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>删除整行</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>删除整列</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>不处理</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QLabel" name="label_outlier_handling">
                 <property name="text">
                  <string>异常值处理：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="combobox_outlier_handling">
                 <item>
                  <property name="text">
                   <string>均值填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>中位数填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>众数填充</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>删除整行</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>删除整列</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>不处理</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QLabel" name="label_categorical_handling">
                 <property name="text">
                  <string>离散值处理：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="combobox_categorical_handling">
                 <item>
                  <property name="text">
                   <string>标签编码（Label Encoding）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>独热编码（One-Hot Encoding）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>二进制编码（Binary Encoding）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>离散哈希（Feature Hashing）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>嵌入方法（Embeddings）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>目标编码（Target Encoding）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>不处理</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QLabel" name="label_data_normalization_handling">
                 <property name="text">
                  <string>数据归一化：</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="combobox_data_normalization_handling">
                 <item>
                  <property name="text">
                   <string>最小-最大归一化（Min-Max Normalization）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>标准化（Z-score Normalization）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>均值归一化（Mean Normalization）</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>不处理</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame_2">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_16">
          <item>
           <widget class="QTabWidget" name="tabWidgetMLRawData">
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true">QTabBar::tab {
    background: lightgray;
    border: 2px solid gray;
    border-bottom-color: #C2C7CB; /* same as the pane color */
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    min-width: 8ex;
    padding: 2px;
}

QTabBar::tab:selected, QTabBar::tab:hover {
    background: white;
    border-color: #9B9B9B;
	font-weight: bold;
}

QTabBar::tab:selected {
    border-bottom-color: white; /* same as pane color */
}
</string>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tab_1">
             <attribute name="title">
              <string>原始数据</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_25">
              <item>
               <widget class="QTableWidget" name="tablewidget_raw_data"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_2">
             <attribute name="title">
              <string>缺失值处理后</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_24">
              <item>
               <widget class="QTableWidget" name="tablewidget_data_missing_handled"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_3">
             <attribute name="title">
              <string>异常值处理后</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_21">
              <item>
               <widget class="QTableWidget" name="tablewidget_data_outlier_handled"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_4">
             <attribute name="title">
              <string>离散值处理后</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_23">
              <item>
               <widget class="QTableWidget" name="tablewidget_data_categorical_handled"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab_5">
             <attribute name="title">
              <string>数据归一化后</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_22">
              <item>
               <widget class="QTableWidget" name="tablewidget_data_normalization_handled"/>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="progressBar">
            <property name="value">
             <number>0</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1031</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
